Jenkinsfile

```diff
@@ -398,6 +398,14 @@ pipeline{
 								sh 'curl -X PUT https://cloud-hypervisor-metrics.azurewebsites.net/api/publishmetrics -H "x-functions-key: $METRICS_PUBLISH_KEY" -T ~/workloads/metrics.json'
 							}
 						}
+						stage ('Run rate-limiter integration tests') {
```

Comment:
This isn't right. You should create a new top-level stage but asking for the same agent. You shouldn't overload the metrics stage.

Reply:
Updated. I now have it running for every PR. Please let me know if you prefer to run only for the main branch.

I was hesitate to add a top-level stage given how 'small' the rate-limiter tests are. I agree overloading metrics stage was not a good idea.

Reply:
Work on the bare metal machines should generally only be run on the main branch.

Reply:
Updated.